x/gamm/keeper/pool_twap_test.go

```diff
@@ -494,10 +494,135 @@ func (suite *KeeperTestSuite) TestPoolTwapSpotPrice() {
+		fn func()
+	}{
+		{
+			// test delete pool twap history with three different pools
+			fn: func() {
+				pool1 := suite.preparePool()
+
+				poolParams := types.PoolParams{
+					SwapFee: sdk.NewDec(0),
+					ExitFee: sdk.NewDec(0),
+				}
+				pool2, err := suite.app.GAMMKeeper.CreatePool(suite.ctx, acc1, poolParams, []types.PoolAsset{
+					{
+						Weight: sdk.NewInt(100),
+						Token:  sdk.NewCoin("foo", sdk.NewInt(5_000_000)),
+					},
+					{
+						Weight: sdk.NewInt(100),
+						Token:  sdk.NewCoin("bar", sdk.NewInt(5_000_000)),
+					},
+				}, "")
+				suite.NoError(err)
+
+				pool3, err := suite.app.GAMMKeeper.CreatePool(suite.ctx, acc1, poolParams, []types.PoolAsset{
+					{
+						Weight: sdk.NewInt(100),
+						Token:  sdk.NewCoin("bar", sdk.NewInt(5_000_000)),
+					},
+					{
+						Weight: sdk.NewInt(100),
+						Token:  sdk.NewCoin("baz", sdk.NewInt(5_000_000)),
+					},
+				}, "")
+				suite.NoError(err)
+
+				foocoin := sdk.NewCoin("foo", sdk.NewInt(10))
+				barcoin := sdk.NewCoin("bar", sdk.NewInt(20))
+
+				ctx := suite.ctx.WithBlockTime(time.Now().Add(time.Minute * 20))
+				_, _, err = suite.app.GAMMKeeper.SwapExactAmountIn(ctx, acc1, pool1, foocoin, "bar", sdk.ZeroInt())
+				suite.Require().NoError(err)
+
+				ctx = suite.ctx.WithBlockTime(time.Now().Add(time.Minute * 30))
+				_, _, err = suite.app.GAMMKeeper.SwapExactAmountIn(ctx, acc1, pool2, foocoin, "bar", sdk.ZeroInt())
+				suite.Require().NoError(err)
+
+				ctx = suite.ctx.WithBlockTime(time.Now().Add(time.Minute * 40))
+				_, _, err = suite.app.GAMMKeeper.SwapExactAmountIn(ctx, acc1, pool1, barcoin, "foo", sdk.ZeroInt())
+				suite.Require().NoError(err)
+
+				ctx = suite.ctx.WithBlockTime(time.Now().Add(time.Minute * 50))
+				_, _, err = suite.app.GAMMKeeper.SwapExactAmountIn(ctx, acc1, pool3, barcoin, "baz", sdk.ZeroInt())
+				suite.Require().NoError(err)
+
+				ctx = suite.ctx.WithBlockTime(time.Now().Add(time.Minute * 55))
+				_, _, err = suite.app.GAMMKeeper.SwapExactAmountIn(ctx, acc1, pool2, foocoin, "bar", sdk.ZeroInt())
+				suite.Require().NoError(err)
+
+				ctx = suite.ctx.WithBlockTime(time.Now().Add(time.Hour*1 + time.Minute*20))
+				_, _, err = suite.app.GAMMKeeper.SwapExactAmountIn(ctx, acc1, pool1, foocoin, "bar", sdk.ZeroInt())
+				suite.Require().NoError(err)
+
+				ctx = suite.ctx.WithBlockTime(time.Now().Add(time.Hour*1 + time.Minute*30))
+				_, _, err = suite.app.GAMMKeeper.SwapExactAmountIn(ctx, acc1, pool3, barcoin, "baz", sdk.ZeroInt())
+				suite.Require().NoError(err)
+
+				ctx = suite.ctx.WithBlockTime(time.Now().Add(time.Hour*1 + time.Minute*40))
+				_, _, err = suite.app.GAMMKeeper.SwapExactAmountIn(ctx, acc1, pool1, foocoin, "bar", sdk.ZeroInt())
+				suite.Require().NoError(err)
+
+				ctx = suite.ctx.WithBlockTime(time.Now().Add(time.Hour*1 + time.Minute*50))
+				_, _, err = suite.app.GAMMKeeper.SwapExactAmountIn(ctx, acc1, pool2, foocoin, "bar", sdk.ZeroInt())
+				suite.Require().NoError(err)
```

Comment:
Can we put this into a loop thats like `num_twap_entries = 12`, `for i := 0; i < num_twap_entries; i++ { {do thing with blocktime being time.Minute * (10 * num_twap_entries) } }`

Reply:
Or do you think I should make the test case itself more simpler?